docs/DesignDocs/SGX_QuoteEx_Integration.md

```diff
@@ -0,0 +1,127 @@
+
+- `sgx_qe_get_target_info(sgx_target_info_t *p_qe_target_info)`
+    - Return the SGX Quoting Enclave (QE) target information, for application enclave to generate its SGX report.
+- `sgx_qe_get_quote_size(uint32_t *p_quote_size)`
+    - Return the size of the buffer needed to hold the SGX ECDSA quote to be generated.
+- `sgx_qe_get_quote(const sgx_report_t *p_app_report, uint32_t quote_size, uint8_t *p_quote)`
+    - Generate an SGX ECDSA quote for the input application enclave SGX report, and return it in the caller-supplied buffer.
+
+### Project Compilation and Linking
+
+As defined in `cmake` configuration file `host/CMakeLists.txt`, for OE SDK built on an SGX platform, the host-side plugin library code is linked with the DCAP static library.
+
+## Proposed Changes
+
+### quote-ex Library API
+
+For generation of SGX evidence in ECDSA and EPID formats, the SGX quote-ex library has the following relevant API functions defined in its [header file](https://github.com/intel/linux-sgx/blob/master/common/inc/sgx_uae_quote_ex.h):
+
+- `sgx_get_supported_att_key_ids(sgx_att_key_id_ext_t *p_att_key_id_list, uint32_t *p_att_key_id_list_size)`
+    - Return the list of supported attestation key IDs (which can be mapped to OE SDK evidence formats) on the current platform.
+    - Note: this function is not yet available in the current release, but will be added in a future release.
+- `sgx_init_quote_ex(const sgx_att_key_id_t* p_att_key_id, sgx_target_info_t *p_qe_target_info, size_t* p_pub_key_id_size, uint8_t* p_pub_key_id);`
+    - Return the SGX Quoting Enclave (QE) target information for the given attestation key ID.
+- `sgx_get_quote_size_ex(const sgx_att_key_id_t *p_att_key_id, uint32_t* p_quote_size)`
+    - Return the size of the buffer needed to hold the quote to be generated for the given attestation key ID.
+- `sgx_get_quote_ex(const sgx_report_t *p_app_report, const sgx_att_key_id_t *p_att_key_id,sgx_qe_report_info_t *p_qe_report_info, uint8_t *p_quote, uint32_t quote_size)`
+    - Generate a quote for the given attestation key ID and application SGX report, and return it in the call-supplied buffer.
+
+As compared to the DCAP library API, the quote-ex library API allows enumeration of supported evidence formats (called attestation key IDs in the API). Otherwise the quote-ex API is similar to the DCAP API, except that every function takes an input attestation key ID in its parameter list.
+
+### Host-side Plugin Library Dynamic Load of the quote-ex or DCAP Library
+
+Between the DCAP and quote-ex libraries:
+- The DCAP library only supports generation of SGX quote in ECDSA-p256 format. With DCAP, the quote generation can be done either in-process, or out-of-process by working with a background service running on the same platform.
```

Comment:
grammar: "SGX quote" -> either "an SGX quote" or "SGX quotes"
Same in 74

Reply:
"SGX quotes"